Scripts/FindScansFromBeamPositions.py

```diff
@@ -0,0 +1,118 @@
+fieldKey=[]
+items=beamPositionLines[0].split(",")
+if items[0] is "fill":
+    fieldKey=items
+    beamPositionLines.pop(0)
+else:
+    #assume may 2016 format
+    fieldKey=stdFirstLine.split(",")
+
+print fieldKey
+
+scanPoints={} # number --> dx, dy, start, stop
+
+if 'set_nominal_B2xingPlane' in fieldKey:
+    print "set_nominal_B2xingPlane in fieldKey"
+
+iScanPoint=0
+curX=0
+curY=0
+for line in beamPositionLines:
+    items=line.split(",")
+    #if items[fieldKey.index('nominal_separation_plane')]  is 'NONE':
+    #    continue
+    try:
+        if args.runs!="":
+            if items[fieldKey.index('run')] not in args.runs:
+                continue
+        dx=float(items[fieldKey.index('set_nominal_B2xingPlane')])-float(items[fieldKey.index('set_nominal_B1xingPlane')])
+        dy=float(items[fieldKey.index('set_nominal_B2sepPlane')])-float(items[fieldKey.index('set_nominal_B1sepPlane')])
+        if dx==0 and dy==0:
+            continue
+        elif abs(dx)<1.e-10 and dy==0:
+            continue
+        elif dx==0 and abs(dy)<1.e-10:
+            continue
+
+        if len(scanPoints.keys())==0:
+            scanPoints[iScanPoint]=[dx,dy,int(items[fieldKey.index('sec')]),-1]
+            curX=dx
+            curY=dy
+        # is new scan?
+        if curX!=dx or curY!=dy: 
+            iScanPoint=iScanPoint+1
+            scanPoints[iScanPoint]=[dx,dy,int(items[fieldKey.index('sec')]),-1]
+            curX=dx
+            curY=dy
+        else:
+            scanPoints[iScanPoint][3]=int(items[fieldKey.index('sec')])
+    except:
+        print "fail",line
+
+print iScanPoint
+scans={}
+iScan=0
+lastTime=0
+for scanPoint in range(iScanPoint):
+    print scanPoint,scanPoints[scanPoint]
+    if len(scans.keys())==0:
+        print "no keys, insert one"
+        if scanPoints[scanPoint][0]==0:
+            scans[iScan]=["Y",scanPoints[scanPoint][2],scanPoints[scanPoint][3]]
+        if scanPoints[scanPoint][1]==0:
+            scans[iScan]=["X",scanPoints[scanPoint][2],scanPoints[scanPoint][3]]
+    else:
+        newScan=False
+        if scans[iScan][0]=="Y" and scanPoints[scanPoint][1]==0:
+            print "Y scan ends... X begins"
+            newScan=True
+        elif scans[iScan][0]=="X" and scanPoints[scanPoint][0]==0:
+            print "X scan ends... Y begins"
+            newScan=True
+        elif int(scanPoints[scanPoint][2])-lastTime>100:
+            print "It's been awhile"
```

Comment:
print "It's been awhile" --> print "^--It's been awhile"

or smth related to indicate that the comment refers to the line above and not the line below